internal/gengapic/doc_file.go

```diff
@@ -79,18 +79,25 @@ func (g *generator) genDocFile(year int, scopes []string, serv *descriptor.Servi
 	p("// The client will use your default application credentials. Clients should be reused instead of created as needed.")
 	p("// The methods of Client are safe for concurrent use by multiple goroutines.")
 	p("// The returned client must be Closed when it is done being used.")
-	p("//")
-	p("// Using the Client")
-	p("//")
-	p("// The following is an example of making an API call with the newly created client.")
-	p("//")
-	// Code block for client using the first method of the service
-	tmpMethod := g.pt
-	g.pt = printer.P{}
-	g.exampleMethodBody(g.opts.pkgName, pbinfo.ReduceServName(serv.GetName(), g.opts.pkgName), serv.GetMethod()[0])
-	snipMethod := g.pt.String()
-	g.pt = tmpMethod
-	g.codesnippet(snipMethod)
+
+	// If the service does not have any methods, then do not generate sample method snippet.
+	if serv.GetMethod() == nil || len(serv.GetMethod()[0].GetName()) == 0 {
+		p("//")
+	} else {
```

Comment:
```suggestion
	p("//")
	if len(serv.GetMethod()) > 0 {
```

The p("//") is a common line to both branches so move it outside of the condition, and invert the condition to only generate pkg level examples if the service has at least one RPC defined.

Reply:
makes sense - I had to use `serv.GetMethod()[0].GetName()` as opposed to just `serve.GetMethod()` because if `[]*MethodDescriptorProto` existed but just had an empty Method object, then the length was still 1 as opposed to 0 (although I'm not sure if that's a valid concern actually - if a service doesn't have any methods defined, would `[]*MethodDescriptorProto` be nil?). 
I inverted the if statement; let me know how it looks.

Reply:
> existed but just had an empty Method object, then the length was still 1 as opposed to 0 (although I'm not sure if that's a valid concern actually

It's not a concern :)

> if a service doesn't have any methods defined, would []*MethodDescriptorProto be nil?

Yep! And in Go, using `len` or `range` on a `nil` slice produces `0` and nothing respectively.

Reply:
ahhh makes sense; okay I've updated the if statement as well the test. Thanks for clarifying!